components/pixels/lib.rs

```diff
@@ -287,7 +287,20 @@ pub struct RasterImage {
     pub frames: Vec<ImageFrame>,
 }
 
-#[derive(Clone, Deserialize, MallocSizeOf, Serialize)]
+fn sensible_delay(delay: Duration) -> Duration {
+    // Very small timeout values are problematic for two reasons: we don't want
+    // to burn energy redrawing animated images extremely fast, and broken tools
+    // generate these values when they actually want a "default" value, so such
+    // images won't play back right without normalization.
+    // https://searchfox.org/firefox-main/rev/c79acad610ddbb31bd92e837e056b53716f5ccf2/image/FrameTimeout.h#35
+    if delay <= Duration::from_millis(10) {
+        Duration::from_millis(100)
+    } else {
+        delay
+    }
+}
+
+#[derive(Clone, Debug, Deserialize, MallocSizeOf, Serialize)]
 pub struct ImageFrame {
     pub delay: Option<Duration>,
```

Comment:
Nit: consider marking this as private to ensure every caller uses the method

Reply:
Maybe we should create `FrameTimeout`, like they did in mozilla: https://searchfox.org/firefox-main/rev/c79acad610ddbb31bd92e837e056b53716f5ccf2/image/FrameTimeout.h#23